admin/helpers/school-home-feature-eligibility-presenter.js

```diff
@@ -0,0 +1,46 @@
+'use strict'
+
+const moment = require('moment')
+
+const config = require('../config')
+const dateService = require('../services/date.service')
+
+const schoolHomeFeatureEligibilityPresenter = {}
+
+/**
+ * Fetch data for familiarisation and live pin generation eligibility
+ * @param checkWindowData
+ * @returns {Object} Eligibility data including flags and relevant datetimes
+ */
+schoolHomeFeatureEligibilityPresenter.getPresentationData = (checkWindowData) => {
```

Comment:
There appears to be a mix of service (business logic) and presenter stuff mixed up here.   The code seems surprisingly dense.  It lacks functions - everything is happening in one block.  nothing can be re-used.

I am not quite sure if there is a specific issue here, but it there seems to be a "code smell" in this block.

Reply:
have to agree.  why not have a method that sets up the raw data (dates etc) . and another that sets the predicate stuff, at the very least.  its mostly unreadable in its current state

Reply:
@jon-shipley I splitted the implementation into several functions. Have another look please.